### PR TITLE
Backport/restore FastSim options with proc mods to override decays with GEN for 10_6 UL

### PR DESCRIPTION
**Pull request by sbein**
To preserve physics, the proc modifier for fixing the long-lived bug is restored, and the option to disregard all FastSim decays in place of GEN decays is now activated with a proc modifier. The main issue was that the backport https://github.com/cms-sw/cmssw/pull/39022 was missing two lines in FastSimProducer.cc conditioning the decay action of the FastSim decayer on the relevant option flag.

### CODE REVIEW

**Comment by sbein** [comment]
please test

**Comment by sbein** [comment]
please test

**Comment by perrotta** [comment]
backport of https://github.com/cms-sw/cmssw/pull/39022
- (Non an identical backport...)

**Comment by sbein** [comment]
@perrotta that's true, but it is backporting the necessary physics option for standard model samples (no FastSim decays) and respecting the no-change policy by using options. Note that releases after >=CMSSW_10_6_31 should not be used for official FastSim productions, so the no-change consistency is with respect to CMSSW_10_6_30.

**Comment by sbein** [comment]
@perrotta this PR is really needed, and it is implementing the physics of the mentioned backport as an option - and restoring the exact behavior of the previous used release, CMSSW_10_6_30.

**Comment by perrotta** [comment]
> @perrotta this PR is really needed, and it is implementing the physics of the mentioned backport as an option - and restoring the exact behavior of the previous used release, CMSSW_10_6_30.

I understand, but you have to fix the clang build error, first!
Then, once you sign it for fastsim, it can be taken into account for merging

**Comment by perrotta** [comment]
> > @perrotta this PR is really needed, and it is implementing the physics of the mentioned backport as an option - and restoring the exact behavior of the previous used release, CMSSW_10_6_30.
> 
> I understand, but you have to fix the clang build error, first! Then, once you sign it for fastsim, it can be taken into account for merging

@sbein please join the release planning meeting this afternoon to support the merging of this PR

**Comment by perrotta** [comment]
please test

**Comment by civanch** [comment]
+1

**Comment by perrotta** [comment]
please test
(old ones have disappeared)

File: FastSimulation/SimplifiedGeometryPropagator/python/fastSimProducer_cff.py [hunk 1]
```diff
@@ -78,4 +77,14 @@
     MaterialEffectsForMuonsInECAL = MaterialEffectsForMuonsInECALBlock.MaterialEffectsForMuonsInECAL,
     MaterialEffectsForMuonsInHCAL = MaterialEffectsForMuonsInHCALBlock.MaterialEffectsForMuonsInHCAL,
     GFlash = FamosCalorimetryBlock.GFlash,
+    fixLongLivedBug = cms.bool(False),
+    useFastSimsDecayer = cms.bool(True),
```
Comment (perrotta) [inline FastSimulation/SimplifiedGeometryPropagator/python/fastSimProducer_cff.py:81]: This changes the previous default

File: FastSimulation/SimplifiedGeometryPropagator/python/fastSimProducer_cff.py [hunk 2]
```diff
@@ -78,4 +77,14 @@
...
 )
+
+from Configuration.ProcessModifiers.fastSimFixLongLivedBug_cff import fastSimFixLongLivedBug
+fastSimFixLongLivedBug.toModify(fastSimProducer, fixLongLivedBug = cms.bool(True))
```
Comment (perrotta) [inline FastSimulation/SimplifiedGeometryPropagator/python/fastSimProducer_cff.py:85]:
```suggestion
fastSimFixLongLivedBug.toModify(fastSimProducer, fixLongLivedBug = True)
```

File: FastSimulation/SimplifiedGeometryPropagator/python/fastSimProducer_cff.py [hunk 3]
```diff
@@ -78,4 +77,14 @@
...
+
+from Configuration.ProcessModifiers.useGenNotFastSimDecays_cff import useGenNotFastSimDecays
+useGenNotFastSimDecays.toModify(fastSimProducer, useFastSimsDecayer = cms.bool(False))
```
Comment (perrotta) [inline FastSimulation/SimplifiedGeometryPropagator/python/fastSimProducer_cff.py:88]:
```suggestion
useGenNotFastSimDecays.toModify(fastSimProducer, useFastSimsDecayer = False)
```

**Review by perrotta — COMMENTED** [review]
FastSim outputs show differences, while I expected that the differences should have been activated by the new modifiers instead, and being disabled by default. Could you please check and comment?

**Comment by sbein** [comment]
@perrotta, indeed, this changes physics with respect to HEAD. This PR undoes an illegal change to physics that has existed since (and including) CMSSW_10_6_31, which was not used for production.

**Comment by sbein** [comment]
+1

**Comment by perrotta** [comment]
Thank you @sbein 

So this fix reverts the wrong setting that was merged with https://github.com/cms-sw/cmssw/pull/39022 and entered the 10_6_X release first with [CMSSW_10_6_31](https://github.com/cms-sw/cmssw/releases/CMSSW_10_6_31) on 25 november 2022.

@cms-sw/pdmv-l2 can you confirm that no (exotic) fastsim production was ever run with 10_6_X releases from CMSSW_10_6_31 to CMSSW_10_6_35? If so, this PR can be (finally!) safely merged.

**Comment by sunilUIET** [comment]
Hi @perrotta, as I can see from the Fastim [campaigns list](https://cms-pdmv.cern.ch/mcm/campaigns?prepid=*FS*&page=-1&shown=63) neither the campaign used CMSSW_10_6_31 to CMSSW_10_6_35

**Comment by perrotta** [comment]
+1

**Comment by perrotta** [comment]
+operations

**Pull request merged**